plutus-book/doc/09-extended.adoc

```diff
@@ -62,83 +66,133 @@ sections of a Haskell program, called Plutus Tx, which are then compiled into
+Only spending from these _script addresses_ can be controlled by the script (whose
+address this is). Let us look at a very small example that omits some of the
+details, but gives the reader an idea of the kind of thing scripts can do.
+Suppose the owner of (non-script) address `a` has promised to pay `x` Ada
+to the first person to guess a word they are thinking of.
+This is a kind of contract, and we can interpret it as a smart contract by
+considering exactly how it controls the transfer of funds. This process
+always starts with a contribution of funds to the contract.
+
+Recall that in the classic model, UTXO entries are of the form
+`(txid, ix) |-> (a,c)`. The pair `(a,c)` has type `TxOut` and is called an output,
+where `a` is the address of the owner of the funds
+(the paying or staking public key, usually), and `c` is the value of Ada in the output.
+
+Roughly, we expect the owner of `a` to pay to the contract `c` Ada, which can
+only be spent if the correct word guess is provided by the participant.
+We also notice here that the secret word cannot be stored in plain text on
+the blockchain, so we must only ever expose the hash of this word.
+So, the contract we want should take a hashed secret word and a plain text
+word guess provided by a participant, and allow him to spend the Ada `c`
+locked at the contract address if the guess is correct.
+To express this, we write in `PlutusTx` the following contract,
+representing the condition under which the transfer of funds can take place:
+
+----
+s = "given the hash hs of a secret word, and a word guess wg,
+check that wg hashes to hs"
+----
+
+The contract is created when the owner of `a` pays to this script (i.e. to the
+address `hash s`) `c` Ada. So, he produces a transaction that contains some inputs spending
+UTXO entries with `a` in the address field, and an output
+
+----
+ix |-> (hash s, c, hs)
+----
+
+Note that the hash of the secret word `hs` is a separate piece of data in the
+output - we will explain
+why in more detail when we give the validator-redeemer-data script model overview.
+Once this transaction is processed, we will have the following entry in the UTXO:
+
+----
+(txid, ix) |-> (hash s, c, hs)
+----
+
+In the classic model, the inputs of a transaction are simply references to
+the UTXO entries which the transaction is spending, so they are
+of the form `(txid, ix)` - just like the keys of the UTXO finite map. In the
+extended model, we have a bit more data in the inputs for Plutus functionality.
+This `(txid, ix)`-key output can then be spent by a wallet that can provide the
+word `wg` that hashes to `hs`. So, it can
+be spent by a transaction containing an input which spends the output
+corresponding to `(txid, ix)` and also carries the correct guess `wg`, so the
+input data is,
+
+----
+(txid, ix, wg)
+----
+
+At a node processing this transaction, then, the computation applying
```

Comment:
```suggestion
At a node processing this transaction the computation applying
```

Reply:
I find that, like you, I naturally tend to write with a lot of subordinate clauses, extra asides, things like this (maybe because that's how I speak). But I'm fairly convinced that they're not helpful in writing. So I highly recommend the (painful) process of just going through all your writing and ruthlessly deleting clauses, adverbs etc. that don't add to the sense.

Of course there's a balance to be struck here with having some writing style, but my experience is that it's much easier to err on the side of too much than too little.

Somewhat exaggerated, but https://youtu.be/8TD-20Mb_7M?t=3161 is a useful example of editing.